schematics/transforms.py

```diff
@@ -426,54 +424,48 @@ def blacklist(*field_list):
 ###
 
 
-def convert(cls, instance_or_dict, trusted_data=None, partial=True, strict=False,
-            mapping=None, app_data=None, context=None):
-    field_converter = lambda field, value, context: field.to_native(value, context)
-    data = import_loop(cls, instance_or_dict, field_converter, trusted_data=trusted_data,
-                       partial=partial, strict=strict, mapping=mapping, app_data=app_data, context=context)
-    return data
+class FieldConverter(object):
```

Comment:
The big idea here is to define an extensible interface. Instead of passing a simple converter function you can pass a callable that may also implement pre- and post-conversion hooks (which allow the application to examine and mutate the dataset as a whole).

For example, I've written a pluggable converter class that uses this facility to export [JSON API](http://jsonapi.org/) -compliant payloads instead of flat dictionaries.

The default implementation is super simple and could indeed be replaced with a plain function. It mainly exists to serve as a model implementation.

(Note that the hooks were implemented later — have a look at the current revision.)